### DISCUSSION TITLE
Middleware/Filter Test

### DISCUSSION THREAD

**Discussion by snowbldr** [TechEmpower/FrameworkBenchmarks#6937] [Ideas]
I would be interested in a test that compared the execution time of middleware or filters in each of the frameworks. 

These are commonly used for authentication, and most frameworks support some version of it. 

Perhaps an "Authenticated JWT Request" test would be appropriate. The executor could craft a jwt and each framework would have to parse a valid jwt and respond to the request. A verification could be added to ensure it returns a 401 when an invalid jwt is passed.

The jwt secret could be injected via mount file or environment variable to each docker container, and generated on each test run.

I'd be interested in hearing your thoughts about this.

**Comment by YVS37** [comment]
Kido

**Comment by YVS37** [comment]
Kido

**Comment by YVS37** [comment]
Kido
